randutil/crypto_test.go

```diff
@@ -45,3 +48,79 @@ func (s *cryptoRandutilSuite) TestCryptoToken(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(b, HasLen, 5)
 }
+
+var (
+	KernelTestUUID        = "1031319a-b661-4c01-aafa-6def8a118944"
+	KernelUUIDPathDefault = randutil.KernelUUIDPath
+)
+
+func (s *cryptoRandutilSuite) TestRandomKernelUUIDNoFile(c *C) {
+	tmpDir := c.MkDir()
+	uuidPath := filepath.Join(tmpDir, "no-file")
+
+	randutil.KernelUUIDPath = uuidPath
+	defer func() {
+		randutil.KernelUUIDPath = KernelUUIDPathDefault
+	}()
+
+	value, err := randutil.RandomKernelUUID()
+	c.Check(value, Equals, "")
+	c.Check(err, ErrorMatches, "*.no such file or directory")
```

Comment:
(niptick) if parts of the path are included here this test also validates that the user gets enough context about the error (i.e. that the `return err` is fine and we may not need to prefix it with something like "cannot generate kernel uuid")